<launch>
  <!-- RealSense avec profondeur alignÃ©e -->
  <include file="$(find doll_head_torso)/launch/camera.launch">
    <arg name="enable_color" value="true"/>
    <arg name="enable_depth" value="true"/>
    <arg name="align_depth" value="true"/>
    <arg name="enable_pointcloud" value="false"/>

    <arg name="color_width"  value="640"/>
    <arg name="color_height" value="480"/>
    <arg name="color_fps"    value="30"/>
    <arg name="depth_width"  value="640"/>
    <arg name="depth_height" value="480"/>
    <arg name="depth_fps"    value="30"/>
  </include>

  <node pkg="doll_head_torso" type="doll_state_node_cord.py" name="doll_state" output="screen">
    <param name="model_path" value="$(find doll_head_torso)/weights/best.pt"/>
    <param name="image_topic" value="/camera/color/image_raw"/>
    <param name="depth_topic" value="/camera/aligned_depth_to_color/image_raw"/>
    <param name="camera_info_topic" value="/camera/color/camera_info"/>
    <param name="use_depth" value="true"/>

    <param name="conf" value="0.33"/>
    <param name="iou"  value="0.58"/>
    <param name="imgsz" value="640"/>
    <param name="device" value="0"/>
    <param name="process_every_n" value="1"/>

    <param name="mask_threshold" value="0.5"/>
    <param name="min_conf_ht" value="0.5"/>
    <param name="ht_union_iou_min" value="0.5"/>

    <param name="overlap_thresh" value="0.22"/>
    <param name="center_dist_factor" value="0.55"/>
    <param name="min_vertical_margin" value="0.05"/>
    <param name="max_head_to_torso_area_ratio" value="0.8"/>

    <param name="min_frames_to_assembled" value="5"/>
    <param name="min_frames_to_disassembled" value="3"/>
    <param name="unknown_decay" value="1"/>

    <param name="publish_debug" value="true"/>
  </node>
</launch>
